src/connection.ts

```diff
@@ -463,6 +468,13 @@ export class Connection {
       response.messages.map((x) => consumer.handle(x))
     })
   }
+
+  private calculateFrameMaxSizeFrom(tuneResponseFrameMax: number) {
+    if (this.frameMax === DEFAULT_UNLIMITED_FRAME_SIZE && tuneResponseFrameMax === DEFAULT_UNLIMITED_FRAME_SIZE) {
+      return 0
+    }
+    return Math.max(1, Math.min(this.frameMax, tuneResponseFrameMax))
```

Comment:
If I'm reading this correctly, if `tuneResponseFrameMax = 0` and `this.frameMax = 100` this expression returns 1 instead of 100.

I suggest something like this in essence:

```suggestion
    if (this.frameMax === DEFAULT_UNLIMITED_FRAME_SIZE) return tuneResponseFrameMax
    if (tuneResponseFrameMax  === DEFAULT_UNLIMITED_FRAME_SIZE) return this.frameMax
return Math.min(this.frameMax, tuneResponseFrameMax)

```